portfolio/src/main/webapp/js/comments.js

```diff
@@ -14,8 +31,42 @@ function initCommentsSystem() {
     event.preventDefault();
     loadComments();
   });
+}
+
+/**
+ * Updates the comments UI according to user authorization. 
+ */
+function applyAuthorizationToUI() {
+  if (commentAuthData.authorized) {
+    const withAuthElements = 
+        Array.from(document.getElementsByClassName("only-display-with-auth"));
+    withAuthElements.forEach((elem) => elem.style.display = "inherit");
+  } else {
+    const withoutAuthElements = Array.from(
+        document.getElementsByClassName("only-display-without-auth"));
+    withoutAuthElements.forEach((elem) => elem.style.display = "inherit");
+    
+    const loginButton = document.getElementById("comment-auth-login");
+    loginButton.href = commentAuthData.loginUrl;
+  }
+}
 
-  loadComments();
+/**
+ * Gets the user's authorization status.
+ */
+function getAuthorization() {
+  return fetch('/auth').then((resp) => {
+    if (resp.ok) {
+      return resp.json();
+    } else {
+      return resp.text().then(
+          text => Promise.reject(`Error ${resp.status}: ${text}`));
+    }
+  }).then((authData) => {
+    console.log("Got comment authorization:");
+    console.log(authData);
+    commentAuthData = authData;
+  });
```

Comment:
I see what you're saying, although keep in mind that `resp.json()` returns a promise, so the change would look like:

`resp.json().then((authData) => {
...
});`

Where it is usually considered bad practice to have multiple depths of promises within a promise chain.

Ironically though I am doing the bad practice in the block below where I am handling the error text. This corresponds to issue #22 where this can be removed when all API errors are handled with JSON.

Reply:
Thoughts?

Reply:
Still thinking about this -- knowing that the above code snippet is a rather common occurrence here (checking if response is okay), I'm wondering if maybe I should create a helper function that handles all of that logic, so the functions that make these fetch calls only need a .then() and .catch().

Reply:
Yeah I know all about callback hell, and promises are nice but don't really solve the problem either.

FWIW async/await is new in JS which helps a bit. And I agree with Joe separating things out into functions may be a good step.

Reply:
Okay. I'll separate this out into functions and see how it looks.

Reply:
I like your idea to separate this out into functions. That's the pattern that http://callbackhell.com/ seems to suggest for handling situations like this.